File: ms-auth/src/auth/infrastructure/messaging/logging-event-publisher.ts
import { Injectable, Logger } from '@nestjs/common';
import { DomainEvent } from '../../../shared/domain-event';
import { EventPublisher } from '../../domain/ports/out/event-publisher';

@Injectable()
export class LoggingEventPublisher implements EventPublisher {
  private readonly logger = new Logger('DomainEvents');

  async publishAll(events: DomainEvent[]): Promise<void> {
    for (const event of events) {
      this.logger.log(
        `${event.eventName()} @ ${event.occurredAt.toISOString()} :: ${JSON.stringify(event)}`,
      );
    }
  }
}

File: ms-doctor/src/main/java/com/encuentratumedico/msdoctor/infrastructure/messaging/LoggingEventPublisher.java
package com.encuentratumedico.msdoctor.infrastructure.messaging;

import com.encuentratumedico.msdoctor.domain.port.out.EventPublisherPort;
import com.encuentratumedico.msdoctor.shared.domain.DomainEvent;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;

// @Component removido: KafkaEventPublisher (@Primary) es el bean activo en producción.
// Activar aquí (volver a añadir @Component sin @Primary) solo para pruebas locales sin Kafka.
public class LoggingEventPublisher implements EventPublisherPort {

    private static final Logger log = LoggerFactory.getLogger("DomainEvents");

    @Override
    public void publishAll(List<? extends DomainEvent> events) {
        if (events == null) return;
        for (DomainEvent event : events) {
            log.info("{} @ {} :: {}", event.eventName(), event.occurredAt(), event);
        }
    }
}
